docs/cookbook/adding-a-new-elasticsearch-index.md

```diff
@@ -0,0 +1,294 @@
+
+```php
+declare(strict_types=1);
+
+namespace App\Model\Category\Elasticsearch;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractExportSubscriber;
+use Shopsys\FrameworkBundle\Component\Elasticsearch\IndexDefinitionLoader;
+use Shopsys\FrameworkBundle\Component\Elasticsearch\IndexRepository;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class CategoryExportSubscriber extends AbstractExportSubscriber
+{
+    /**
+     * @param \App\Model\Category\Elasticsearch\CategoryExportScheduler $categoryExportScheduler
+     * @param \Doctrine\ORM\EntityManagerInterface $entityManager
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\IndexRepository $indexRepository
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\IndexDefinitionLoader $indexDefinitionLoader
+     * @param \App\Model\Category\Elasticsearch\CategoryIndex $index
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(
+        CategoryExportScheduler $categoryExportScheduler,
+        EntityManagerInterface $entityManager,
+        IndexRepository $indexRepository,
+        IndexDefinitionLoader $indexDefinitionLoader,
+        CategoryIndex $index,
+        Domain $domain
+    ) {
+        parent::__construct($categoryExportScheduler, $entityManager, $indexRepository, $indexDefinitionLoader, $index, $domain);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::RESPONSE => [
+                ['exportScheduledRows', 0],
+            ],
+        ];
+    }
+}
+```
+
+#### CategoryIndex::getExportDataForIds()
+
+To finish partial exports we need to implement the last unimplemented method in `CategoryIndex` to return all categories by their identifiers.
+We may also use already existing method from `CategoryRepository`.
+
+```diff
+    public function getExportDataForIds(int $domainId, array $restrictToIds): array
+    {
+-       // TODO: Implement getExportDataForIds() method.
++       $categories = $this->categoryRepository->getCategoriesByIds($restrictToIds);
++
++       $domainConfig = $this->domain->getDomainConfigById($domainId);
++       $locale = $domainConfig->getLocale();
++       foreach ($categories as $category) {
++           $result[$category->getId()] = $this->convertToElastic($category, $domainId, $locale);
++       }
++       return $result;
+    }
+```
+
+## Conclusion
+
+We have created a new index category into elasticsearch. We are able to fill it with data (by a cron or immediately after row is changed).
+From now you are able to use elasticsearch as a data source (data providing functionality is needed to be implemented) instead PostgreSQL which will save your performance.
```

Comment:
```suggestion
From now you are able to use elasticsearch as a data source (data providing functionality is needed to be implemented) instead of PostgreSQL which will improve the performance of your application.
```